File: MyMindNotes/app/src/main/java/com/android/mymindnotes/SituationFragment.java
package com.android.mymindnotes;

import androidx.activity.result.ActivityResult;
import androidx.activity.result.ActivityResultCallback;
import androidx.activity.result.ActivityResultLauncher;
import androidx.activity.result.contract.ActivityResultContracts;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.app.Activity;
import android.graphics.Point;
import android.os.Bundle;
import android.view.Display;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.android.mymindnotes.databinding.ActivitySituationFragmentBinding;
import com.android.mymindnotes.model.UserDiary;

public class SituationFragment extends Fragment {
    ActivitySituationFragmentBinding binding;

    // 화면 크기에 따른 글자 크기 조절
    int standardSize_X, standardSize_Y;
    float density;

    // 화면 크기에 따라서 글자 사이즈 조절하기
    public Point getScreenSize(Activity activity) {
        Display display = activity.getWindowManager().getDefaultDisplay();
        Point size = new Point();
        display.getSize(size);

        return size;
    }
    public void getStandardSize() {
        Point ScreenSize = getScreenSize(getActivity());
        density  = getResources().getDisplayMetrics().density;

        standardSize_X = (int) (ScreenSize.x / density);
        standardSize_Y = (int) (ScreenSize.y / density);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = ActivitySituationFragmentBinding.inflate(getLayoutInflater());
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        binding.content.setText(getArguments().getString("situation"));
        getStandardSize();
        binding.content.setTextSize((float) (standardSize_X / 23));
    }

    public void refreshData(UserDiary diary) {
        binding.content.setText(diary.getSituation());
    }

    public static SituationFragment newInstance(Bundle bundle) {
        SituationFragment fragment = new SituationFragment();
        fragment.setArguments(bundle);
        return fragment;
    }
}

File: MyMindNotes/app/src/main/java/com/android/mymindnotes/SituationFragment.kt
package com.android.mymindnotes

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.android.mymindnotes.data.retrofit.model.diary.UserDiary
import com.android.mymindnotes.databinding.ActivitySituationFragmentBinding
import dagger.hilt.android.AndroidEntryPoint
import javax.inject.Inject

@AndroidEntryPoint
class SituationFragment @Inject constructor(

) : Fragment() {
    lateinit var binding: ActivitySituationFragmentBinding

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        binding = ActivitySituationFragmentBinding.inflate(layoutInflater)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding.content.text = arguments?.getString("situation")
    }

    fun refreshData(diary: UserDiary) {
        binding.content.text = diary.situation
    }

    companion object {
        fun newInstance(bundle: Bundle?): SituationFragment {
            val fragment = SituationFragment()
            fragment.arguments = bundle
            return fragment
        }
    }
}